client/java/generator/src/main/java/io/openlineage/client/JavaPoetGenerator.java

```diff
@@ -230,6 +215,138 @@ private void generateTypes(TypeSpec.Builder containerTypeBuilder) {
 
   }
 
+  private void generateInterface(TypeSpec.Builder containerTypeBuilder, ObjectResolvedType type) {
+    TypeSpec.Builder interfaceBuilder = TypeSpec.interfaceBuilder(type.getName())
+        .addModifiers(STATIC, PUBLIC);
+
+    ///////////////////////////////////////////
+    // Default implementation to deserialize //
+    ///////////////////////////////////////////
+    if (type.getName().endsWith("Facet") && !type.getName().equals("BaseFacet")) {
+      // adding the annotation to the interface to have a default implementation
+      interfaceBuilder.addAnnotation(AnnotationSpec.builder(JsonDeserialize.class)
+          .addMember("as", CodeBlock.of("Default" + type.getName() + ".class"))
+          .build());
+
+      TypeSpec.Builder classBuilder = TypeSpec.classBuilder("Default" + type.getName())
+          .addModifiers(STATIC, PRIVATE, FINAL);
+      classBuilder.addSuperinterface(ClassName.get(PACKAGE, containerClassName, type.getName()));
+
+      MethodSpec.Builder constructor = MethodSpec.constructorBuilder()
+          .addModifiers(PRIVATE);
+      constructor.addAnnotation(JsonCreator.class);
+      List<String> fieldNames = new ArrayList<String>();
+      for (ResolvedField f : type.getProperties()) {
+        classBuilder.addField(getTypeName(f.getType()), f.getName(), PRIVATE, FINAL);
+        fieldNames.add(f.getName());
+        if (isSchemaURL(f)) {
+          setSchemaURLField(type, constructor, f);
+        } else {
+          addConstructorParameter(constructor, f);
+        }
+        MethodSpec getter = getter(f)
+            .addModifiers(PUBLIC)
+            .addCode("return $N;", f.getName())
+            .build();
+        classBuilder.addMethod(getter);
+      }
+
+      // additionalFields
+      if (type.hasAdditionalProperties()) {
+        addAdditionalProperties(type, classBuilder, constructor);
+      }
+      classBuilder.addMethod(constructor.build());
+      containerTypeBuilder.addType(classBuilder.build());
+    }
+    ///////////////////////////////
```

Comment:
Maybe just split the above into its own method?

Reply:
done